app/models/tenant.rb

```diff
@@ -0,0 +1,149 @@
+class Tenant < ActiveRecord::Base
+  validates :subdomain, uniqueness: true
+
+  after_create :create_tenant
+  after_destroy :destroy_tenant
+
+  private
+    def create_tenant
+      unless subdomain == "public"
+        Apartment::Tenant.create(subdomain)
+        Apartment::Tenant.switch(subdomain) do
+          admin = User.create!(username: 'admin', email: 'admin@consul.dev', password: '12345678', password_confirmation: '12345678', confirmed_at: Time.current, terms_of_service: "1")
+          admin.create_administrator
+          populate_settings
+        end
+      end
+    end
+
+    def destroy_tenant
+      Apartment::Tenant.drop(subdomain)
+    end
+
+    def populate_settings
```

Comment:
:thinking: why do we need to populate the DB through the model? Can't we do that with a Rake task?

Reply:
We need to load data configuration when creating a new tenant, but I agree with you that it's not the best way to do it. I'll try to do it as you suggested.